Add few clients to handle more Hubspot api endpoints

jisson opened the pull request:
Add few clients in order to handle the following hubspot api endpoints:
 - properties
 - property groups
 - assocations
 - pipelines

Also add few methods to existing clients.

Commit:
Add few clients to handle more Hubspot api endpoints

Add few clients in order to handle the following hubspot api endpoints:
 - properties
 - property groups

Also add few methods to existing clients and define the version of [fire](google/python-fire#120) to `0.1.3` in `requirements.txt` as the last version dont have the `helputils` package.

jisson commented on hubspot3/base.py:
```diff
@@ -95,7 +111,31 @@ def _get_path(self, subpath):
 
     def _prepare_request_auth(self, subpath, params, data, opts):
         if self.api_key:
-            params["hapikey"] = params.get("hapikey") or self.api_key
```
@Jenselme 

I would be happy to have your opinion on this part of the code. I am not sure why we made this change and I don't know yet if It should be removed.

Jenselme replied:
It feels weird and I'm not sure I understand what it does (mostly why we need to catch `AttributeError`). Did you try to remove these changes and see if it still works with our use cases?

jisson replied:
After few tests, it seems to work nice without those changes. We are always using the `api_key` so it seems ok.

The only thing that scare me is this comment regarding `urlencode` but I cannot remember now a reason to this comment.

I think that it is preferable to remove it now and see later if we really need to update this method. Right now I am unable to justify this change to the owner of the library.

Jenselme commented on hubspot3/associations.py:
```diff
@@ -0,0 +1,163 @@
+"""
```
If that's something we added it may not be useful any more. See https://github.com/jpetrucciani/hubspot3/blob/master/hubspot3/crm_associations.py

jisson replied:
Thanks. It seems that it is the same thing for `pipelines`. I am looking at it

Jenselme commented on hubspot3/associations.py:
```diff
@@ -0,0 +1,163 @@
+            )
+
+        return self._call(
+            "", method="PUT", data={
```
Why the empty string here?

Jenselme commented on hubspot3/base.py:
```diff
@@ -110,7 +150,9 @@ def _prepare_request(self, subpath, params, data, opts, doseq=False, query=""):
             query = "&" + query
         url = opts.get("url") or "/{}?{}{}".format(
-            self._get_path(subpath), urllib.parse.urlencode(params, doseq), query
+            self._get_path(subpath),
```
This doesn't look necessary.

Jenselme reviewed (commented):
The lib evolved a lot. Pay attention not to submit things that already exist ;-) I have small remarks but it looks ready to be submitted to me (may without the weird part you flagged).

Commit:
Add get_deals_pipeline_by_id

Add `get_deals_pipeline_by_id` to `crm_pipelines`, allowing to retrieve a deal from its id.

Commit:
Add methods to CRMAssociationsClient

Add few methods (as shortcuts) to the `CRMAssociationsClient`.
